app/src/main/kotlin/io/sweers/catchup/ui/CoilImagesPlugin.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2019. Zac Sweers
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.sweers.catchup.ui
+
+import android.content.Context
+import android.graphics.drawable.Drawable
+import android.text.Spanned
+import android.widget.TextView
+import coil.ImageLoader
+import coil.api.load
+import coil.target.Target
+import io.noties.markwon.AbstractMarkwonPlugin
+import io.noties.markwon.MarkwonConfiguration
+import io.noties.markwon.MarkwonSpansFactory
+import io.noties.markwon.image.AsyncDrawable
+import io.noties.markwon.image.AsyncDrawableLoader
+import io.noties.markwon.image.AsyncDrawableScheduler
+import io.noties.markwon.image.DrawableUtils
+import io.noties.markwon.image.ImageSpanFactory
+import org.commonmark.node.Image
+import java.util.HashMap
+
+class CoilImagesPlugin(context: Context, imageLoader: ImageLoader) : AbstractMarkwonPlugin() {
+
+  private val coilAsyncDrawableLoader = CoilAsyncDrawableLoader(context, imageLoader)
+
+  override fun configureSpansFactory(builder: MarkwonSpansFactory.Builder) {
+    builder.setFactory(Image::class.java, ImageSpanFactory())
+  }
+
+  override fun configureConfiguration(builder: MarkwonConfiguration.Builder) {
+    builder.asyncDrawableLoader(coilAsyncDrawableLoader)
+  }
+
+  override fun beforeSetText(textView: TextView, markdown: Spanned) {
+    AsyncDrawableScheduler.unschedule(textView)
+  }
+
+  override fun afterSetText(textView: TextView) {
+    AsyncDrawableScheduler.schedule(textView)
+  }
+
+  // TODO cancellation isn't supported right now. Would need to make this be able to pass a
+  //  lifecycle to the load
+  private class CoilAsyncDrawableLoader(
+    private val context: Context,
+    private val imageLoader: ImageLoader
+  ) : AsyncDrawableLoader() {
+    private val cache = HashMap<AsyncDrawable, Target>(2)
+
+    override fun load(drawable: AsyncDrawable) {
+      val target = AsyncDrawableTarget(drawable)
+      cache[drawable] = target
+      imageLoader.load(context, drawable.destination) {
```

Comment:
`load` returns a `RequestDisposable`, which could be used to cancel the Request.

Reply:
True, I started down this road but got stuck on something at the time. Will revisit